Add support in kernel to send append flag for ipv4 route append

diff --git a/net/ipv4/fib_trie.c b/net/ipv4/fib_trie.c
index 17d4d47..948ee5c 100644
--- a/net/ipv4/fib_trie.c
+++ b/net/ipv4/fib_trie.c
@@ -1194,6 +1194,7 @@ int fib_table_insert(struct fib_table *tb, struct fib_config *cfg)
 	u32 key, mask;
 	int err;
 	struct leaf *l;
+	unsigned int flags = 0;
 
 	if (plen > 32)
 		return -EINVAL;
@@ -1304,6 +1305,8 @@ int fib_table_insert(struct fib_table *tb, struct fib_config *cfg)
 
 		if (!(cfg->fc_nlflags & NLM_F_APPEND))
 			fa = fa_first;
+		else
+			flags |= NLM_F_APPEND;
 	}
 	err = -ENOENT;
 	if (!(cfg->fc_nlflags & NLM_F_CREATE))
@@ -1338,7 +1341,7 @@ int fib_table_insert(struct fib_table *tb, struct fib_config *cfg)
 
 	rt_cache_flush(cfg->fc_nlinfo.nl_net, -1);
 	rtmsg_fib(RTM_NEWROUTE, htonl(key), new_fa, plen, tb->tb_id,
-		  &cfg->fc_nlinfo, 0);
+		  &cfg->fc_nlinfo, flags);
 succeeded:
 	return 0;
 
